src/transformers/generation/configuration_utils.py

```diff
@@ -627,30 +629,36 @@ def validate(self, is_init=False):
                 )
 
         # check watermarking arguments
-        if self.watermarking_args is not None:
+        if self.watermarking_config is not None:
```

Comment:
Cool validation!

suggestion: this code block could become a function within `WatermarkingConfig` (e.g. `WatermarkingConfig.validate()`), and here we would simply call the function. That way, the class also contains its own validation, as opposed to being done in an external class :)